app/dashboard/models.py

```diff
@@ -927,12 +973,32 @@ def amount_in_whole_units(self):
     def receive_url(self):
         if self.web3_type == 'yge':
             return self.url
-
+        if self.web3_type == 'v3':
+            return self.receive_url_for_recipient
+        if self.web3_type != 'v2':
+            raise Exception
+        
         pk = self.metadata.get('priv_key')
         txid = self.txid
         network = self.network
         return f"{settings.BASE_URL}tip/receive/v2/{pk}/{txid}/{network}"
 
+    @property
+    def receive_url_for_recipient(self):
+        if self.web3_type != 'v3':
+            raise Exception
+
+        key = self.metadata['reference_hash_for_receipient']
+        return f"{settings.BASE_URL}tip/receive/v3/{key}/{self.txid}/{self.network}"
+
+    @property
+    def receive_url_for_funder(self):
+        if self.web3_type != 'v3':
+            raise Exception
```

Comment:
Opposed to raising an `Exception` here, can we handle it by returning an empty string?

Reply:
i think that Exception is the right route, because a tip that's not a v3 tip shouldn't have separate funder/contrib URLs, so this method should not be called